src/FitsViewerExt.js

```diff
@@ -178,13 +177,14 @@ export class FitsViewerWidget extends Widget {
 }
 
 
-function tellLabToloadFileToServer( filename, firefly) {
-    const {fetchUrl}= firefly.util;
-    const UL_URL= window.document.location.href+ `/sendToFirefly?path=${filename}`;
-    const options = { method: 'GET' };
-    return fetchUrl(UL_URL,options,false, false)
-        .catch( e => {
-            console.log('Got Error from upload request');
+function tellLabToLoadFileToServer( path, firefly) {
+    const {searchParams,origin,pathname}= new URL(window.document.location.href);
+    searchParams.append('path',path);
+    const slashMaybe= pathname.endsWith('/') ? '' : '/';
```

Comment:
For this part, I was expecting something that would truncate the starting url anywhere a question mark appears. I saw a suggestion to use `split('?')`.

Reply:
URL handles that.